### PR TITLE
Go SDK

### PR DESCRIPTION
**Pull request by laves**
Adding Go support for Rhino and associated demos

### CODE REVIEW

File: demo/go/go.mod [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module rhinodemo
+
+go 1.16
+
+require (
+	github.com/Picovoice/rhino/binding/go v0.0.0-20210521013428-88b4dc4b0f91
```
Comment (dbartle) [inline demo/go/go.mod:6]: see if this can be tagged

File: demo/go/filedemo/rhino_file_demo.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+	}
+	inputAudioPath, _ := filepath.Abs(*inputAudioPathArg)
+	f, err := os.Open(inputAudioPath)
+	if err != nil {
+		log.Fatalf("Unable to find or open input audio at %s", inputAudioPath)
+	}
+	defer f.Close()
+
+	wavFile := wav.NewDecoder(f)
+	if !wavFile.IsValidFile() || wavFile.BitDepth != 16 || wavFile.SampleRate != 16000 || wavFile.NumChans != 1 {
```
Comment (dbartle) [inline demo/go/filedemo/rhino_file_demo.go:48]: excellent

File: demo/go/README.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+## Installation
+
+MicDemo uses [malgo](https://github.com/gen2brain/malgo) for cross-platform audio capture. It requires `cgo`, which on Windows may mean that you need to install a gcc compiler like [Mingw](http://mingw-w64.org/doku.php) to build it properly. 
+
+## Usage
+
+NOTE: The working directory for the following go commands is:
+
+```console
+rhino/demo/go
```
Comment (dbartle) [inline demo/go/README.md:14]: nitpick but add "cd" since this is what they're supposed to type in their prompt

**Review by dbartle — COMMENTED** [review]
Looks solid. Investigate the tagging solution discussed and see if we can get versions working with the mono-repo.

Would prefer to keep unrelated things out of the CR even if they're just comment/README updates. (fine to keep them for this CR but for future, avoid).

File: binding/go/README.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+Rhino is:
+
+* using deep neural networks trained in real-world environments.
+* compact and computationally-efficient, making it perfect for IoT.
+* self-service. Developers and designers can train custom models using [Picovoice Console](https://picovoice.ai/console/).
+
+## Compatibility
+
+- Go 1.16+
+- Runs on Linux (x86_64), macOS (x86_64) and Windows (x86_64)
```
[the remark this answers is not in the document]
Comment (laves) [reply binding/go/README.md:33]: not yet, that's a stretch goal

File: demo/go/go.mod (hunk 1 above)
Comment (laves) [reply demo/go/go.mod:6]: will do after merge

File: binding/go/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (dbartle) [reply binding/go/README.md:33]: the rpi files are checked in so if it doesn't support it those need to get moved out

**Pull request merged**